src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -35,6 +35,16 @@ internal class PsesInternalHost : PSHost, IHostSupportsInteractiveSession, IRuns
     {
         internal const string DefaultPrompt = "> ";
 
+        // TODO: Make the __psEditorServices prefix shared (it's used elsewhere too).
+        private const string ShellIntegrationScript = @"
+# Define fake PSConsoleHostReadLine so VS Code's script can execute it to get the user's input.
+$global:__psEditorServices_userInput = """";
+function global:PSConsoleHostReadLine { $global:__psEditorServices_userInput }
+
+# Execute VS Code's built-in shell integration script.
+try { . ""$(code --locate-shell-integration-path pwsh)"" } catch {}
```

Comment:
My main concern is not knowing if `code` will reliably in `PATH`...

Reply:
You should be able to get the executable from the extension host's `process`. Also keep in mind there are other flavors like `code-insiders` and `code-exploration`.

Reply:
Perhaps the best thing to do would be to get the path to the script in the client and pass that along to the server to use here.

Reply:
Fixed it up.